contrib/zip/utils/utils_zip.cc

```diff
@@ -0,0 +1,322 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "contrib/zip/utils/utils_zip.h"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include "absl/cleanup/cleanup.h"
+#include "contrib/zip/sandboxed.h"
+
+constexpr uint64_t kFileMaxSize = 1024 * 1024 * 1024;  // 1GB
+
+#define ZIP_FL_ENC_GUESS 0
+#define ZIP_FL_OVERWRITE 8192u
+
+LibZip::~LibZip() {
+  if (IsOpen()) {
+    api_.zip_close(&(*zip_)).IgnoreError();
```

Comment:
Use `std::unique_ptr<>::get()` instead of deref and taking its address. Also elsewhere in this file.

Reply:
Fixed.